src/Symfony/Component/Cache/Adapter/FilesystemAdapter.php

```diff
@@ -147,7 +147,7 @@ protected function doSave(array $values, $lifetime)
     private function getFile($id, $mkdir = false)
     {
         $hash = str_replace('/', '-', base64_encode(md5($id, true)));
```

Comment:
Using md5 which has known collisions with just 64 bytes is already unsafe. But ignoring the case of base64_encode will make it even worse. This basically means it unsafe to use untrusted cache keys for the cache component. This needs to be highlighted.